File: UNIDAD 1/01_SOA_CON_BDD/002. REST_JAVA_EUREKABANK_GR10/02. CLICON/EUREKA_REST_CLICON_GR10/src/ec/edu/monster/controlador/EurekaController.java
package ec.edu.monster.controlador;

import ec.edu.monster.servicio.EurekaService;
import ec.edu.monster.modelo.Movimiento;
import java.util.List;

/**
 * Controller to coordinate between console input and service layer.
 * @author MATIAS
 */
public class EurekaController {

    private final EurekaService service;

    public EurekaController() {
        this.service = new EurekaService();
    }

    public List<Movimiento> consultarMovimientos(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        List<Movimiento> movimientos = service.leerMovimientos(cuenta);
        if (movimientos.isEmpty()) {
            throw new Exception("No se encontraron movimientos para la cuenta.");
        }
        return movimientos;
    }

    public void registrarDeposito(String cuenta, double importe, String codEmp) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        service.registrarDeposito(cuenta, importe, codEmp);
    }

    public void registrarRetiro(String cuenta, double importe, String codEmp) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        service.registrarRetiro(cuenta, importe, codEmp);
    }

    public void realizarTransferencia(String cuentaOrigen, String cuentaDestino, double importe, String codEmp) throws Exception {
        if (cuentaOrigen == null || cuentaOrigen.trim().isEmpty() || cuentaDestino == null || cuentaDestino.trim().isEmpty()) {
            throw new Exception("Los números de cuenta son requeridos.");
        }
        if (cuentaOrigen.equals(cuentaDestino)) {
            throw new Exception("Las cuentas de origen y destino deben ser diferentes.");
        }
        if (importe <= 0) {
            throw new Exception("El importe debe ser mayor que 0.");
        }
        if (codEmp == null || codEmp.trim().isEmpty()) {
            throw new Exception("El código de empleado es requerido.");
        }
        service.realizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp);
    }

    public double verificarSaldo(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        return service.verificarSaldo(cuenta);
    }

    public double obtenerCostoMovimiento(String cuenta) throws Exception {
        if (cuenta == null || cuenta.trim().isEmpty()) {
            throw new Exception("El número de cuenta es requerido.");
        }
        return service.obtenerCostoMovimiento(cuenta);
    }

    public boolean iniciarSesion(String usuario, String clave) throws Exception {
        if (usuario == null || usuario.trim().isEmpty() || clave == null || clave.trim().isEmpty()) {
            throw new Exception("Usuario y contraseña son requeridos.");
        }
        return service.iniciarSesion(usuario, clave) != null;
    }
}

File: UNIDAD 1/01_SOA_CON_BDD/002. REST_DOTNET_EUREKABANK_GR10/02. CLICON/EurekaBankConsoleClient/EurekaBankConsoleClient/Controlador/EurekaController.cs
using EurekaBankConsoleClient.Models;
using EurekaBankConsoleClient.Servicio;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EurekaBankConsoleClient.Controlador
{
    public class EurekaController
    {
        private readonly EurekaService _service;

        public EurekaController()
        {
            _service = new EurekaService();
        }

        public async Task<List<Movimiento>> ConsultarMovimientos(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            List<Movimiento> movimientos = await _service.TraerMovimientos(cuenta);
            if (movimientos == null || movimientos.Count == 0)
            {
                throw new Exception("No se encontraron movimientos para la cuenta.");
            }
            return movimientos;
        }

        public async Task<bool> RegistrarDeposito(string cuenta, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            return await _service.RegistrarDeposito(cuenta, importe, codEmp) == 1;
        }

        public async Task<bool> RegistrarRetiro(string cuenta, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            double costo = await _service.ObtenerCostoMovimiento(cuenta);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return await _service.RegistrarRetiro(cuenta, importe, codEmp) == 1;
        }

        public async Task<bool> RealizarTransferencia(string cuentaOrigen, string cuentaDestino, double importe, string codEmp)
        {
            if (string.IsNullOrWhiteSpace(cuentaOrigen) || string.IsNullOrWhiteSpace(cuentaDestino))
            {
                throw new Exception("Los números de cuenta son requeridos.");
            }
            if (importe <= 0)
            {
                throw new Exception("El importe debe ser mayor que 0.");
            }
            if (string.IsNullOrWhiteSpace(codEmp))
            {
                throw new Exception("El código de empleado es requerido.");
            }
            double costo = await _service.ObtenerCostoMovimiento(cuentaOrigen);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return await _service.RealizarTransferencia(cuentaOrigen, cuentaDestino, importe, codEmp) == 1;
        }

        public async Task<double> VerificarSaldo(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            double saldo = await _service.VerificarSaldo(cuenta);
            if (saldo == -1.0)
            {
                throw new Exception("Error al verificar el saldo.");
            }
            return saldo;
        }

        public async Task<double> ObtenerCostoMovimiento(string cuenta)
        {
            if (string.IsNullOrWhiteSpace(cuenta))
            {
                throw new Exception("El número de cuenta es requerido.");
            }
            double costo = await _service.ObtenerCostoMovimiento(cuenta);
            if (costo == -1.0)
            {
                throw new Exception("Error al obtener el costo del movimiento.");
            }
            return costo;
        }

        public async Task<bool> IniciarSesion(string usuario, string clave)
        {
            if (string.IsNullOrWhiteSpace(usuario) || string.IsNullOrWhiteSpace(clave))
            {
                throw new Exception("Usuario y contraseña son requeridos.");
            }
            var user = await _service.IniciarSesion(usuario, clave);
            return user != null;
        }
    }
}
